fedbiomed/researcher/secagg/_secagg_context.py

```diff
@@ -539,6 +543,111 @@ def _create_payload_specific(self) -> Tuple[Union[dict, None], bool]:
+
+        Returns:
+            True if this context can be used with this element, False if not.
+        """
+        return matching_parties_servkey(context, self._parties)
+
+    def _create_payload_specific(self, context) -> Tuple[Union[dict, None], bool]:
+        """Researcher payload for creating Additive Secret Sharing secagg context element
+
+        Args:
+            context: specific secagg element context
+
+        Returns:
+            A tuple of a `context` and a `status` for the server key context element
+        """
+        # context = {'key': context,
+        #            'share': self._share.reconstruct()}
+
+        self._secagg_manager.add(self._secagg_id, self._parties, context, self._experiment_id)
+
+        logger.debug(
+            f"Additive Secret Sharing context successfully created for researcher_id='{environ['ID']}' "
+            f"secagg_id='{self._secagg_id}'")
+
+        return context, True
+
+    def setup(self) -> bool:
+        msg = ResearcherMessages.format_outgoing_message({
+            'researcher_id': self._researcher_id,
+            'secagg_id': self._secagg_id,
+            'element': self._element.value,
+            'experiment_id': self._experiment_id,
+            'parties': self._parties,
+            'command': 'secagg-additive-ss-setup-request',
+        })
+
+        return self._secagg_round(msg, True, self._create_payload)
+
+    def _secagg_round_specific(
+            self,
+            msg: Message,
+            payload: Callable,
+    ) -> Tuple[dict, dict[str, bool]]:
+        """Negotiate secagg context element action with defined parties for a MPSPDZ key exchange.
+
+        Args:
+            msg: message sent to the parties during the round
+            payload: function that holds researcher side payload for this round. Needs to return
+                a tuple of `context` and `status` for this action
+
+        Returns:
+            A tuple of
+                - a dict containing the context describing this secagg context element
+                - a dict where key/values are node ids/boolean with success status for secagg on each party
+
+        Raises:
+            FedbiomedSecaggError: some parties did not answer before timeout or answered error
+            FedbiomedSecaggError: local payload did not complete before timeout
+        """
+
+        # Federated request should stop if any error occurs
+        policies = [StopOnDisconnect(timeout=30), StopOnError(), StopOnTimeout(timeout=120)]
+        # Ensure that reply from each node was received
+        status = {node_id: False for node_id in self._parties}
+
+        # one controller per secagg object to prevent any file conflict
+
+        context = {'share': []}
+        with self._requests.send(msg, self._parties, policies) as fed_request:
+            replies = fed_request.replies()
+            errors = fed_request.errors()
+
+            status = {rep.node_id: rep.success for rep in replies.values()}
```

Comment:
I know that it is in the scope of this PR but I recommend to make `success` field optional, and ignore checking this value. `fed_request.errors()` collects `ErroMessage`s so on the node side we can implement it in a way if there there is an error during setup node just return `ErrorMessage`.

Reply:
Ok I can fix this

Reply:
Actually, i was wondering if we can have this edge case: if all the nodes doesnot reply before timeout? Or if for some reason the message is badly formatted and errors are not registered inside the fed_request? DO you think this could ever happen?

Reply:
If the errors are badly formated fed request will fail due to time out. It is already checked in the code `fed_request.has_stopped_any()`

Reply:
Any if an error happened during the `Round` run? 
eg if for some reason breakpoint is not saved?
 `self._send_round_reply(success=False, message="Can't save new node state.")` 
For what I understood the node only sends `TrainReply`

Reply:
>And if a node to node connection fails? will we get a ErrorMessage?

Yes